src/ddmd/dsymbolsem.d

```diff
@@ -1621,7 +1621,8 @@ extern(C++) final class Semantic3Visitor : Visitor
         if (funcdecl.parameters)
```

Comment:
I'd say change this line to:
```
if (funcdecl.parameters && !funcdecl.errors)
```
because inferring scope is not necessary after the function has errors in it.